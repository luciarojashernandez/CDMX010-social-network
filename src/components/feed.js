export const feed = `
<div id="feedScreen" class="container2">
  <header class="menu">
      <img src="images/logo app2.png" class="logoMenu">
      <section class="iconHeader">
        <input type="image" id="alcaldiasIcon" class="alcaldiasIcon" src="images/alcaldias.png">
        <input type="image" id="btnSignOut" class="menuIcon" src="images/signoutIcon.png">
      </section>
  </header>
  <div id="btnPost">
    <input type="button" id="goPostScreen" class="orangeButton" value="CREAR NUEVO POST">
    </div>
  <div id="postContainer" class="postContainer"></div>
</div>
    `;

export const buildPost = async (onGetPost) => {
  const postContainer = document.querySelector('#postContainer');
  await onGetPost((postCollection) => {
    postContainer.innerHTML = '';
    postCollection.forEach((element) => {
      console.log('Estamos en el callback de consol loge');
      const infoPost = element.data();
      infoPost.id = element.id;
      const numberLike = infoPost.like.length;
      postContainer.innerHTML += `
      <div class="singlePost">
      <p id="location">Ubicación: ${infoPost.alcaldias}</p>
      <p>${infoPost.post}</p>
      <div class="optionsBtn"> 
        <h5>${numberLike}</h5>
        <input type="image" id="likeIcon" class="likeIcon ${infoPost.id}" data-id="${infoPost.id}" src="images/likeIcon.png">
        <input type="image" id="editIcon" class="editIcon btnEdit" data-id="${infoPost.id}" src="images/editIcon.png">
        <input type="image" id="deleteIcon" class="deleteIcon btnDelete" data-id="${infoPost.id}" src="images/deleteIcon.png">
      </div>
      </div>
        `;
    });
  });
};

export const removePost = (deletePost, postId) => {
  // eslint-disable-next-line no-restricted-globals
  const option = confirm('¿Estás seguro de querer eliminar el post?');
  if (option === true) {
    deletePost(postId).then(() => {
      console.log(`post${postId}borrado con exito`);
    });
  }
};

export const funcLike = (postId, getPost, editPost) => {
  const emailStorage = localStorage.getItem('emailStorage');
  // const colorBtn = null;
  getPost(postId).then((post) => {
    const emailData = post.data().like;
    let likeActive = false;
    if (emailData.length !== 0) {
      // eslint-disable-next-line no-restricted-syntax
      emailData.forEach((email) => {
        if (email === emailStorage) {
          likeActive = true;
        }
      });
    }
    if (likeActive === false) {
      emailData.push(emailStorage);
      editPost(postId, {
        like: emailData,
      });
      // .then(() => {
      //   // colorBtn = document.querySelector(`input[data-id=${postId}]`);
      //   colorBtn = document.querySelector(`.${postId}`);
      //   colorBtn.setAttribute('src', 'images/likebrown.png');
      // });
    } else {
      const emailPosition = emailData.indexOf(emailStorage);
      emailData.splice(emailPosition, 1);
      editPost(postId, {
        like: emailData,
      });
      // .then(() => {
      //   colorBtn = document.querySelector(`.${postId}`);
      //   colorBtn.setAttribute('src', 'images/likeIcon.png');
      // });
    }
  });
};
