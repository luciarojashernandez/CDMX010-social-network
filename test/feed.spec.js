// importamos la funcion que vamos a testear
import { buildPost, removePost } from '../src/components/feed.js';

describe('feed test', () => {
  beforeEach(() => {
    document.body.innerHTML = '<div id="postContainer"></div>';
  });
  // eslint-disable-next-line jest/no-focused-tests
  test.only('debería renderizar buildPost', async () => {
    const target = document.getElementById('postContainer');
    const postDB = [
      {
        data: () => ({
          alcaldias: 'obregon',
          id: 'G6cHwWgAOGnfidMokUeJ',
          like: '["susana23@gmail.com"]',
          post: 'hola',
        }),
      },
    ];
    // eslint-disable-next-line max-len
    const onGetPost = jest.fn().mockImplementation((callback) => Promise.resolve(postDB).then(callback));
    // const firebase = { onGetPost };
    await buildPost(onGetPost);
    expect(target.innerHTML).toMatchSnapshot();
  });

  test('removePost debería eliminar un post', async () => {
    const target = document.getElementById('root');
    const postDB = [
      {
        alcaldias: 'obregon',
        id: 'G6cHwWgAOGnfidMokUeJ',
        like: '["susana23@gmail.com"]',
        post: 'hola',
      },
      {
        alcaldias: 'iztacalco',
        id: 'ZRlmWqNuzOeyUxYBPGcv',
        like: '["hola@gmail.com"]',
        post: 'gorditas de chicharron',
      },
    ];
    const deletePost = jest.fn().mockImplementation(() => Promise.resolve(postDB));
    await removePost(deletePost, 'G6cHwWgAOGnfidMokUeJ');
    expect(target.innerHTML).toMatchSnapshot();
  });

  // it('debería ser una función', () => {
  //   expect(typeof buildPost).toBe('function');
  // });
});
