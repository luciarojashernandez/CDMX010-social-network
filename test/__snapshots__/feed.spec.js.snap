// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`feed test debería renderizar buildPost 1`] = `
"
      <div class=\\"singlePost\\">
      <p id=\\"location\\">Ubicación: obregon</p>
      <p>hola</p>
      <div class=\\"optionsBtn\\"> 
        <h5>22</h5>
        <input type=\\"image\\" id=\\"likeIcon\\" class=\\"likeIcon undefined\\" data-id=\\"undefined\\" src=\\"images/likeIcon.png\\">
        <input type=\\"image\\" id=\\"editIcon\\" class=\\"editIcon btnEdit\\" data-id=\\"undefined\\" src=\\"images/editIcon.png\\">
        <input type=\\"image\\" id=\\"deleteIcon\\" class=\\"deleteIcon btnDelete\\" data-id=\\"undefined\\" src=\\"images/deleteIcon.png\\">
      </div>
      </div>
        "
`;
